### PR TITLE
Update CarrierService update and create method to account UPS endpoints

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Update CarrierService update and create method to account UPS endpoints
# Testing

Add unit tests

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by nwithan8 — APPROVED** [review]

File: lib/EasyPost/Constant/Constants.php [hunk 1]
```diff
@@ -17,10 +17,15 @@ abstract class Constants
     // Validation
     const CARRIER_ACCOUNT_TYPES_WITH_CUSTOM_WORKFLOWS = [
         'FedexAccount',
-        'UpsAccount',
         'FedexSmartpostAccount'
     ];
 
+    const UPS_ACCOUNT_TYPES_WITH_CUSTOM_WORKFLOWS = [
```
Comment (Justintime50) [inline lib/EasyPost/Constant/Constants.php:23]: suggestion: Prefer this called something like `UPS_OAUTH_ACCOUNT_TYPES` to indicate how they're different than the normal custom workflows

File: lib/EasyPost/Service/CarrierAccountService.php [hunk 2]
```diff
@@ -43,7 +43,16 @@ public function all(mixed $params = null): mixed
...
     public function update(string $id, mixed $params): mixed
     {
-        return self::updateResource(self::serviceModelClassName(self::class), $id, $params);
+        $carrierAccount = self::retrieve($id);
+        $carrierAccountType = $carrierAccount['type'];
+        if (in_array($carrierAccountType, Constants::UPS_ACCOUNT_TYPES_WITH_CUSTOM_WORKFLOWS, true)) {
+            $updateUrl = 'upsOauthRegistration';
+            $params = [self::selectTopLayerKey($carrierAccountType) => $params];
+        } else {
+            $updateUrl = 'carrierAccount';
```
Comment (Justintime50) [inline lib/EasyPost/Service/CarrierAccountService.php:52]: These are not URLs and should not be named as such. They become a URL later but are actually class names here. Also, wouldn't they be title cased?

```suggestion
            $className = 'UpsOauthRegistration';
            $params = [self::selectTopLayerKey($carrierAccountType) => $params];
        } else {
            $className = 'CarrierAccount';
```

Comment (jchen293) [reply lib/EasyPost/Service/CarrierAccountService.php]: I don't think it's case sensitive, but I have uppercased the first character

**Review by Justintime50 — APPROVED** [review]

File: lib/EasyPost/Service/CarrierAccountService.php (hunk 2 above)
Comment (Justintime50) [reply lib/EasyPost/Service/CarrierAccountService.php]: Class names must be capitalized, otherwise it's not a true class name.

**Pull request merged**